internal/repository/employee/employee_test.go

```
@@ -0,0 +1,422 @@
+package employee
+
+import (
+	"context"
+	"fmt"
+	conf "github.com/dilyara4949/employees-api/internal/config"
+	"github.com/dilyara4949/employees-api/internal/database"
+	"github.com/dilyara4949/employees-api/internal/domain"
+	"github.com/dilyara4949/employees-api/internal/repository/position"
+	"log"
+	"reflect"
+	"testing"
+)
+
+func InitData(posRepo domain.PositionsRepository, empRepo domain.EmployeesRepository) {
+	positions := []domain.Position{
+		{
+			ID:     "1",
+			Name:   "name1",
+			Salary: 1,
+		},
+		{
+			ID:     "2",
+			Name:   "name2",
+			Salary: 2,
+		},
+		{
+			ID:     "3",
+			Name:   "name3",
+			Salary: 3,
+		},
+	}
+
+	employees := []domain.Employee{
+		{
+			ID:         "1",
+			FirstName:  "firstname1",
+			LastName:   "lastname1",
+			PositionID: "1",
+		},
+		{
+			ID:         "2",
+			FirstName:  "firstname2",
+			LastName:   "lastname2",
+			PositionID: "2",
+		},
+		{
+			ID:         "3",
+			FirstName:  "firstname3",
+			LastName:   "lastname3",
+			PositionID: "3",
+		},
+	}
+
+	for _, p := range positions {
+		_, _ = posRepo.Create(context.Background(), p)
+	}
+	for _, e := range employees {
+		_, _ = empRepo.Create(context.Background(), e)
+	}
+}
+
+func DeleteData(posRepo domain.PositionsRepository, empRepo domain.EmployeesRepository) {
+	positions := []domain.Position{
+		{
+			ID:     "1",
+			Name:   "name1",
+			Salary: 1,
+		},
+		{
+			ID:     "2",
+			Name:   "name2",
+			Salary: 2,
+		},
+		{
+			ID:     "3",
+			Name:   "name3",
+			Salary: 3,
+		},
+	}
+
+	employees := []domain.Employee{
+		{
+			ID:         "1",
+			FirstName:  "firstname1",
+			LastName:   "lastname1",
+			PositionID: "1",
+		},
+		{
+			ID:         "2",
+			FirstName:  "firstname2",
+			LastName:   "lastname2",
+			PositionID: "2",
+		},
+		{
+			ID:         "3",
+			FirstName:  "firstname3",
+			LastName:   "lastname3",
+			PositionID: "3",
+		},
+	}
+
+	for _, e := range employees {
+		_ = empRepo.Delete(context.Background(), e.ID)
```

I would suggest not discarding errors, and properly return them instead. And in case of an error the test should fail.

---

fixed